File: hydrograph.engine/hydrograph.engine.cascading/src/main/java/hydrograph/engine/adapters/InputRedShiftAdapter.java
package hydrograph.engine.adapters;

import hydrograph.engine.adapters.base.InputAdapterBase;
import hydrograph.engine.cascading.assembly.InputRedshiftAssembly;
import hydrograph.engine.cascading.assembly.base.BaseComponent;
import hydrograph.engine.cascading.assembly.infra.ComponentParameters;
import hydrograph.engine.core.component.generator.InputRedshiftEntityGenerator;
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent;

public class InputRedShiftAdapter extends InputAdapterBase{

	InputRedshiftAssembly redShiftComponent;
	InputRedshiftEntityGenerator entityGenerator;
	
	public InputRedShiftAdapter(TypeBaseComponent baseComponent) {
		entityGenerator=new InputRedshiftEntityGenerator(baseComponent);
	}
	
	@Override
	public BaseComponent getAssembly() {
		return redShiftComponent;
	}

	@Override
	public void createAssembly(ComponentParameters componentParameters) {
		redShiftComponent=new InputRedshiftAssembly(entityGenerator.getEntity(), componentParameters);
		
	}

}

File: hydrograph.engine/hydrograph.engine.spark/src/main/scala/hydrograph/engine/spark/components/adapter/InputRedshiftAdapter.scala
package hydrograph.engine.spark.components.adapter

import hydrograph.engine.core.component.generator.InputRedshiftEntityGenerator
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent
import hydrograph.engine.spark.components.InputRedshiftComponent
import hydrograph.engine.spark.components.adapter.base.InputAdatperBase
import hydrograph.engine.spark.components.base.InputComponentBase
import hydrograph.engine.spark.components.platform.BaseComponentParams


class InputRedshiftAdapter(typeBaseComponent: TypeBaseComponent) extends InputAdatperBase {

  var inputRedshiftEntityGenerator: InputRedshiftEntityGenerator = null
  var inputRedshiftComponent: InputRedshiftComponent = null;

  override def getComponent(): InputComponentBase = inputRedshiftComponent

  override def createGenerator(): Unit = {
    inputRedshiftEntityGenerator = new InputRedshiftEntityGenerator(typeBaseComponent)
  }

  override def createComponent(baseComponentParams: BaseComponentParams): Unit = {
    inputRedshiftComponent = new InputRedshiftComponent(inputRedshiftEntityGenerator.getEntity, baseComponentParams)
  }
}
